features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/ResourcesSerializer.kt

```diff
@@ -17,29 +17,38 @@ import androidx.annotation.VisibleForTesting
 import androidx.annotation.WorkerThread
 import com.datadog.android.api.InternalLogger
 import com.datadog.android.core.internal.utils.executeSafe
-import com.datadog.android.sessionreplay.internal.recorder.base64.Cache.Companion.DOES_NOT_IMPLEMENT_COMPONENTCALLBACKS
-import com.datadog.android.sessionreplay.internal.utils.Base64Utils
+import com.datadog.android.sessionreplay.internal.async.DataQueueHandler
+import com.datadog.android.sessionreplay.internal.async.NoopDataQueueHandler
+import com.datadog.android.sessionreplay.internal.recorder.resources.Cache.Companion.DOES_NOT_IMPLEMENT_COMPONENTCALLBACKS
 import com.datadog.android.sessionreplay.internal.utils.DrawableUtils
 import com.datadog.android.sessionreplay.model.MobileSegment
+import java.util.Collections
 import java.util.concurrent.ExecutorService
 import java.util.concurrent.LinkedBlockingDeque
 import java.util.concurrent.ThreadPoolExecutor
 import java.util.concurrent.TimeUnit
 
 @Suppress("TooManyFunctions")
-internal class Base64Serializer private constructor(
+internal class ResourcesSerializer private constructor(
```

Comment:
I think this class should not be called a serializer ? I think the scope change into something different. For me this class has too many responsibilities now and it is so hard to track the logic inside. We are using it to check if the resource already exists in cache was already solved before and also in the same time to update the queue and mark the wireframe as resolved. These are 2 different responsibilities and should be extracted in separated components. Also there is nothing that suggests that this is a serializer for me here.

Reply:
I do not mind if we do this in another PR but I really think we need to do it now and simplify this logic a bit more.

Reply:
It's renamed in the next pr to ResourceResolver and some responsibilities are extracted out